contentcuration/contentcuration/test_settings.py

```diff
@@ -4,6 +4,11 @@
 
 DEBUG = True
 
+if RUNNING_TESTS:
+    # if we're running tests, run Celery tests synchronously so tests won't complete before the process
+    # is finished.
+    CELERY_TASK_ALWAYS_EAGER = True
```

Comment:
Are we using the eager mode of celery to do the unit testing? I remember from the celery doc, it recommends not using the eager mode to do unit tests: http://docs.celeryproject.org/en/latest/userguide/testing.html
But I may misunderstand this part here. Do you mind explaining this a bit more? Thank you!

Reply:
You're correct about what the docs say. The important difference here is that the docs are concerned with testing the Celery tasks themselves, in which case you won't get an accurate test unless you run async. (e.g. a task will run differently in foreground than background, so foreground testing doesn't allow you to accurately test real-world behavior)

In our case, though, we are primarily testing the communication between the Celery Task and our Task API / state. (i.e. that certain events trigger certain changes in the Task record) In that case, running it async makes us unable to know when to check that the Task DB has been updated, and requires us to wait in tests. Running sync, we can reliably run these checks.

Ultimately, we will want end2end integration tests to fully test the real world behavior, but for unit tests I still wanted to have some limited ability to test the functionality since unit tests are quick and can be run for each build.

Reply:
thank you! It makes sense to me now to have the tests running sync. When I initially tried to write the tests, I also found this eager mode but was confused by the celery doc I put above.. Thanks again for your detailed explanation! I really appreciate it

Reply:
Sure thing, thanks for all your hard work on the model! :)